src/hitchhiker/tree/messaging.cljc

```diff
@@ -0,0 +1,252 @@
+(ns hitchhiker.tree.messaging
```

Comment:
Fantastic! I really appreciate this new view on the diffs--it has given me a ton of confidence that your implementation is correct.